### ISSUE TITLE
Unable to get client IP when running on overlay network

### ISSUE THREAD

**Issue by marccampbell** [containous/traefik#614]
This may be an issue with Docker overlay networks and not Traefik specifically. 

When running traefik as a docker 1.12 service and exposing a port, it connects to both the specified network from the service file and the built-in ingress network.  When the upstream service receives requests forwarded from Traefik, the X-Forwarded-For header contains an IP address from the overlay network, not the actual client address.

**Comment by osixia** [comment]
May be related to:
https://github.com/kubernetes/kubernetes/issues/10921

**Comment by Starefossen** [comment]
I am experiencing the same behaviour when working with Kubernetes and Træfɪk locally through Minikube.

**Comment by Starefossen** [comment]
I had some hopes that the Kubernetes v1.4 would solve this; ref http://kubernetes.io/docs/user-guide/load-balancer/#loss-of-client-source-ip-for-external-traffic. But so far I have not had any success.

**Comment by osixia** [comment]
I now run kubernetes 1.4.0 and nothing changed :(

**Comment by emilevauge** [comment]
Ping @vdemeester, any idea on this?

**Comment by Rio** [comment]
Just ran in to this today while slowly moving away from our old load balancing infra.
Solution for us was to use an external load balancer to send traffic to NodePorts for the service... not ideal.

**Comment by Starefossen** [comment]
@Rio are you using Træfik at all in your setup?

**Comment by Rio** [comment]
Yes, we plan use it as our main ingress point. We've been already using it for services like grafana, staytus and other services that are not dependent on source ip.

**Comment by viet** [comment]
I still need this new feature, because I run a startup that provides TCP connection only for Minecraft players, there's no IP forwarding via HTTP header. Getting users IP via network layer is always better.

**Comment by tzapu** [comment]
i think this should be possible now using 
--publish mode=host,target=80,published=80
and maybe --global 

i however was unable to access traefik when publish is set to mode=host...

**Comment by farko88** [comment]
any update for this?

**Comment by mbrucher** [comment]
Noticed the same problem, spent several weeks figuring out this problem!!

**Comment by NitroBAY** [comment]
https://stackoverflow.com/questions/44639958/nginx-behind-traefik-docker-swarm-mode-real-ip

**Comment by samip5** [comment]
This is present on non-kubernets deployments.

**Comment by georg90** [comment]
I am wondering why this is not getting more interest. Any swarm cluster is (IMHO) not usable with traefik..
We can route and forward but we do not know where the request came from, this kills the traefik own `ipWhitelist` feature as well as anything in the backend (e.g. fail2ban).

Did anybody manage to work around this issue? Currently I am split between using host mode and having the ip but loosing the ingress network or having no option to limit access by source address / exposing the client ip to my apps (e.g. adguard would be much more useful with client IP)

**Comment by CumpsD** [comment]
I am trying to figure this out as well. I have downstream DNS services and they all receive the IP of the traefik instance, while I need the IP of the client

**Comment by munrad** [comment]
Maybe this solution will help in some way
https://github.com/newsnowlabs/docker-ingress-routing-daemon
